scripts/cloudformation/cloudformation_deploy.sh

```diff
@@ -51,13 +52,23 @@ else
     )
 fi
 
-if [ "$ACTION" == "create-stack" ]; then
-  echo "Waiting on cloudformation stack ${CFN_STACK_NAME} $ACTION completion..."
-  aws cloudformation wait stack-create-complete --stack-name ${CFN_STACK_NAME}
-else
-  echo "Waiting on cloudformation stack ${CFN_STACK_NAME} $ACTION completion..."
-  aws cloudformation wait stack-update-complete --stack-name ${CFN_STACK_NAME}
-fi
 
-aws cloudformation describe-stacks --stack-name ${CFN_STACK_NAME} | jq '.Stacks[0]'
-echo "Finished cloudfromation action $ACTION successfully !!!"
\ No newline at end of file
+until echo "$STATUS" | egrep -q 'CREATE_COMPLETE|UPDATE_COMPLETE|COMPLETE|FAILED|DELETE_IN_PROGRESS'; 
+do   
+  aws cloudformation describe-stack-events --stack-name "${CFN_STACK_NAME}" --query 'StackEvents[?contains(ResourceStatus,`CREATE_IN_PROGRESS`)].[LogicalResourceId, ResourceStatus, ResourceType, ResourceStatusReason]';
+  aws cloudformation describe-stack-events --stack-name "${CFN_STACK_NAME}" --query 'StackEvents[?contains(ResourceStatus,`FAILED`)].[LogicalResourceId, ResourceStatus, ResourceType, ResourceStatusReason]';
+  sleep 60; 
```

Comment:
Does this mean cf will always take at minimum 1 min? What if there's no changes? Is there a reason this shouldn't be a much smaller number like 10 or even 5?

Reply:
Using 5 makes it extremely verbose. I will set it to 10.